gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala

```diff
@@ -808,8 +797,10 @@ case class ColumnarOverrideRules(session: SparkSession)
 
   private def postOverrides(): List[SparkSession => Rule[SparkPlan]] =
     List(
-      (_: SparkSession) => TransformPostOverrides(isAdaptiveContext),
-      (s: SparkSession) => VanillaColumnarPlanOverrides(s)
+      (_: SparkSession) => TransformPostOverrides(),
+      (s: SparkSession) => VanillaColumnarPlanOverrides(s),
+      (s: SparkSession) => RemoveGlutenTableCacheColumnarToRow(s),
+      (s: SparkSession) => RemoveTopmostColumnarToRow(s, isAdaptiveContext)
```

Comment:
Shall we apply `RemoveGlutenTableCacheColumnarToRow` and `RemoveTopmostColumnarToRow` first ? To reduce the tree depth before going to transform.

Reply:
It's feasible but I actually be inclined to put it a little bit back since there might be some rules adding C2R / R2C by themselves after `TransformPostOverrides`. E.g., `VanillaColumnarPlanOverrides` is adding C2Rs and R2Cs (of course, it may not add topmost C2R). So `RemoveTopmostColumnarToRow` can guard against all bad cases to prevent the AQE error.